pkg/util/names/names.go

```diff
@@ -121,3 +121,19 @@ func IsReservedNamespace(namespace string) bool {
 func GenerateImpersonationSecretName(clusterName string) string {
 	return fmt.Sprintf("%s-impersonator", clusterName)
 }
+
+// GeneratePropagationPolicyName generates the propagationPolicy name
+func GeneratePropagationPolicyName(namespace, name, gvk string) string {
+	hash := fnv.New32a()
+	hashutil.DeepHashObject(hash, gvk)
+
+	return strings.ToLower(fmt.Sprintf("%s-%s-%s", namespace, name, rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))))
```

Comment:
No `namespace` is require, we can put it in hash, like `hashutil.DeepHashObject(hash, namespace+gvk)`.

And we also limit the length of `hash` value, because the name of resources seem not to over 63 char length.

Reply:
> No `namespace` is require, we can put it in hash, like `hashutil.DeepHashObject(hash, namespace+gvk)`.
> 
> And we also limit the length of `hash` value, because the name of resources seem not to over 63 char length.

some confused : all functions on generate names on package names  do not limit the length of return value like `GenerateWorkName`  ,  show we fix the problem together or later

Reply:
@lonelyCZ help to review here

Reply:
Thanks. I will review it today.

/assign